### PR TITLE
Modded map weight field audit

### PR DESCRIPTION
**Pull request by Procyonae**
#### Summary
None

#### Purpose of change

A map's "weight" field defaults to 1000 if undefined and is only necessary to include when there are multiple OMTs with the same id.

#### Describe the solution

Removes unnecessary weight fields (if there are variants with one having default weight it's still included so the difference can be seen more easily at a glance)
Disables vanilla road in aftershock exoplanet
Reorganises order of maps so same id maps are next to each other
Minor doc change to make usage of the field clearer
Doesn't touch No Hope bc ew

#### Describe alternatives you've considered

Leaving it be

#### Additional context

I'll probably do a vanilla version when I feel particularly masochistic

### CODE REVIEW

**Comment by Procyonae** [comment]
Before this is merged, an aok on the Xedra change to vampire lodges would be good, basically the way it was working you could get the ground floor of the regular version with the vampire basement etc and I didn't think it looked intentional, if it is I'll just revert it.

**Comment by Procyonae** [comment]
Honestly having thought on it a bit being able to have multiple variants of the same om_terrain id is really redundant (it can easily be done with nests that make it more obvious that it's being done and with more functionality), the only use it has is being able to add variants of an existing map in mods but if only one map with each om_terrain was allowed you could override and extend parts of the map in a much friendlier way, for example if you just wanted to spawn extra mobs or items on a map you could just access the existing map through its id and add what you want without needing a duplicate of the map. This would make mods that alter stuff in maps but not maps themselves way more maintainable, quicker to make additions for etc.

File: data/mods/Defense_Mode/mapgen/mansion.json [hunk 1]
```diff
@@ -368,6 +360,58 @@
       "place_loot": [ { "item": "lawn_dart", "x": [ 7, 16 ], "y": [ 18, 22 ], "chance": 40, "repeat": [ 1, 2 ] } ]
     }
   },
+  {
+    "type": "mapgen",
+    "method": "json",
+    "om_terrain": [ "dm_mansion_t2d" ],
+    "weight": 1000,
```
Comment (I-am-Erk) [inline data/mods/Defense_Mode/mapgen/mansion.json:367]: Should this weight field be getting removed if it's just the default val?

File: data/mods/Defense_Mode/mapgen/mansion.json [hunk 2]
```diff
@@ -1574,12 +1551,63 @@
       }
     }
   },
+  {
+    "type": "mapgen",
+    "method": "json",
+    "om_terrain": [ "dm_mansion_+2d" ],
+    "weight": 800,
```
Comment (I-am-Erk) [inline data/mods/Defense_Mode/mapgen/mansion.json:1558]: Similar question, since you seem to be removing all the other weights

File: data/mods/Xedra_Evolved/mapgen/overmap_special.json [hunk 3]
```diff
@@ -283,5 +283,25 @@
     "occurrences": [ 5, 100 ],
     "flags": [ "UNIQUE", "WILDERNESS" ],
     "spawns": { "group": "GROUP_GENIUS_HOMULLUS", "population": [ 10, 30 ], "radius": [ 1, 3 ] }
+  },
+  {
+    "type": "overmap_special",
+    "id": "Vampire Hunting Lodge",
```
Comment (I-am-Erk) [inline data/mods/Xedra_Evolved/mapgen/overmap_special.json:289]: I'm not sure where this entry got migrated from, but a big audit PR like this that's focused on mapgen weights should probably not also be doing this kind of cleanup in other files. It's just too much to track and check for errors.

File: data/mods/desert_region/mapgen/waterbody.json [hunk 4]
```diff
@@ -16,194 +16,34 @@
...
-              "river_c_not_ne",
-              "river_c_not_nw",
-              "river_c_not_se",
-              "river_c_not_sw",
-              "bridge",
-              "bridgehead_ground"
-            ]
-          }
-        },
+        { "else_chunks": [ "shore_north_west" ], "x": 0, "y": 0, "flags": { "north_west": [ "BRIDGE", "LAKE", "RIVER" ] } },
```
Comment (I-am-Erk) [inline data/mods/desert_region/mapgen/waterbody.json:24]: Same thing, this is a good change but should probably be done in another PR, along with a bunch of the related cleanups.

**Review by I-am-Erk — DISMISSED** [review]
I've stopped at around 80% through for the reasons listed. Feel free to offer a counterargument, but currently you're doing "weight" updates in a ton of files (great) and also some unrelated structural cleanup, and I think it's just a bit much. As a person inclined to do the same thing, I'd suggest splitting the two up to keep the blame history clean and so that this is a bit easier to safely review and make sure nothing's been missed in the copy and pasting

File: data/mods/Defense_Mode/mapgen/mansion.json (hunk 1 above)
Comment (Procyonae) [reply data/mods/Defense_Mode/mapgen/mansion.json:367]: "(if there are variants with one having default weight it's still included so the difference can be seen more easily at a glance)"
I appreciate that field is doing nothing it just helps users not familiar with the default to see the relative weights and for anyone to easier identify it as a om_terrain id with variants without having to rely on noticing the other map(s)

File: data/mods/Xedra_Evolved/mapgen/overmap_special.json (hunk 3 above)
Comment (Procyonae) [reply data/mods/Xedra_Evolved/mapgen/overmap_special.json:289]: It's not migrating from anywhere, the vampire lodge maps currently spawn as part of the regular hunting lodge but as far as I can tell whoever did it thought they replace the vanilla maps bc currently you can get the vanilla basement with the vampire ground floor etc so I just made it it's own special so the whole thing spawns together. This is kind of weight related but I can pull it out to it's own PR if you'd still prefer that. (It won't need id migration or anything bc the old ids are still in use by vanilla)

File: data/mods/desert_region/mapgen/waterbody.json (hunk 4 above)
Comment (Procyonae) [reply data/mods/desert_region/mapgen/waterbody.json:24]: That's fair other than that I have no related clean ups I care to do and have 0 interest in doing anything specific with desert_region so it would just be it's own tiny PR. Again if that's desired I'll do it tho.

**Comment by Procyonae** [comment]
The reason I did this PR for context was to track down if there were any mods using huge weights to "replace" vanilla maps (of which there's only 2) and I thought I might as well remove the weight fields doing nothing bc peeps seem to think it does something it doesn't. I appreciate the other minor changes I've made have dubious connections at best to the name of the PR but I don't want to make 10 tiny PRs that I have to keep track of and waste tests and reviewers time on etc. I appreciate it muddys blame history a bit but the commits with other changes contain comments with a brief description of the change which I hoped would be enough, honestly this is much more organised than alot of my other PRs where I've also contained unrelated changes that I spotted while doing other stuff.

File: data/mods/Xedra_Evolved/mapgen/overmap_special.json (hunk 3 above)
Comment (Maleclypse) [reply data/mods/Xedra_Evolved/mapgen/overmap_special.json:289]: Yep, I thought I'd fixed that but apparently never did.  Thanks for catching this!

File: data/mods/Xedra_Evolved/mapgen/safehouse.json [hunk 5]
```diff
@@ -4,7 +4,6 @@
     "om_terrain": "safe_house",
     "//": "used by safe_house_basement",
     "type": "mapgen",
-    "weight": 10,
```
Comment (Maleclypse) [inline data/mods/Xedra_Evolved/mapgen/safehouse.json:7]: Ok so if the weight here is not useful how do I make sure that this is a very rare house spawn? I understand this likely means this has been spawning more than I'd like already.

Comment (Procyonae) [reply data/mods/Xedra_Evolved/mapgen/safehouse.json:7]: The weight of 5 in https://github.com/CleverRaven/Cataclysm-DDA/blob/394f235c205b8b72f39d86ee79b66a2e0f92c6d3/data/mods/Xedra_Evolved/regional_overlay.json#L5
is what defines how often it spawns which comparing it to
https://github.com/CleverRaven/Cataclysm-DDA/blob/master/data/json/regional_map_settings.json#L834-L1006
makes it pretty rare

**Comment by Procyonae** [comment]
Drafted while I make PRs for the unrelated stuff and then I'll remove them here

**Comment by Procyonae** [comment]
Redrafted bc I forgot to do dinomod still and am stopping for today

**Comment by Procyonae** [comment]
@I-am-Erk can you re-review thanks
